### ISSUE TITLE
feat(forge): `forge flatten`

### ISSUE THREAD

**Issue by brockelmore** [gakonst/foundry#312]
Add an `flatten` command that would grab a contract and its dependencies and print to stdout. With the other issues around `verify-contract`, this would provide at the very least users the ability to get forge's best guess at what the compiler is doing with the standard-json input and allow a user to tweak and manually verify if needed. Also I imagine this will surface issues around `verify-contract` for debugging it.

i.e.:
```solidity
pragma solidity ^0.8.0;

import "./ContractDep.sol";

contract Contract is ContractDep {
}
```

would print out:
```solidity
pragma solidity ^0.8.0;
contract ContractDep {
     uint256 myVar;
}
contract Contract is ContractDep {
}
```

**Comment by gakonst** [comment]
I think this is typically called `flatten`? Agreed otherwise that it can be useful.

**Comment by lucas-manuel** [comment]
Yeah this would be analogous to `hevm flatten`

**Comment by brockelmore** [comment]
sure i was basing it off of `cargo expand` didnt know `hevm flatten` existed. Agree flatten makes more sense

**Comment by mattsse** [comment]
closed #506
